### PR TITLE
Add doxygen into gribcreate.f, Part of #55

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Part of #55 
Doxygen has been added into gribcreate.f
It has been tested on
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/gribcreate_8f.html

### CODE REVIEW

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by Hang-Lei-NOAA** [comment]
Looks like Brian's approval doesn't let the system give me the pass.

**Comment by BrianCurtis-NOAA** [comment]
I do not have write access. That would be why.

**Comment by Hang-Lei-NOAA** [comment]
Okay, Thanks!

**Comment by GeorgeGayno-NOAA** [comment]
> Okay, Thanks!

I have write access. I can review.

**Review by BrianCurtis-NOAA — APPROVED** [review]

File: src/gribcreate.f [hunk 1]
```diff
@@ -1,60 +1,50 @@
...
+!>    - listsec1(10) Reference Time - Minute
+!>    - listsec1(11) Reference Time - Second
+!>    - listsec1(12) Production status of data (Code Table 1.3)
+!>    - listsec1(13) Type of processed data (Code Table 1.4)
+!>    @param[out] ierr Error return code.
+!>    - 0 no error
+!>    - 1 Tried to use for version other than GRIB Edition 2
+!>
+!>    @note This routine is intended for use with routines
+!>    addlocal(), addgrid(), addfield(), and gribend() to create a
```
Comment (GeorgeGayno-NOAA) [inline src/gribcreate.f:41]: @Hang-Lei-NOAA I see doxygen links to addlocal() and gridend(). But not to addgrid() and addfield(). Does anyone know why?

Comment (Hang-Lei-NOAA) [reply src/gribcreate.f:41]: It was the warnings from other two code. I will get it fixed in next commit.
Please see this, it has been fixed.
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/gribcreate_8f.html

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**